src/validation.cpp

```diff
@@ -5681,7 +5681,8 @@ util::Result<void> ChainstateManager::ActivateSnapshot(
             return util::Error{strprintf(_("The base block header (%s) is part of an invalid chain."), base_blockhash.ToString())};
         }
 
-        if (Assert(m_active_chainstate->GetMempool())->size() > 0) {
+        auto mempool{m_active_chainstate->GetMempool()};
```

Comment:
nit: The type doesn't seem so complex that it has to be auto

```suggestions
CTxMemPool* mempool{m_active_chainstate->GetMempool()};
```